software-mansion/react-native-reanimated
# Format Babel plugin test inputs

## Summary

This PR changes the way how JavaScript code snippets are embedded in Reanimated Babel plugin unit tests so that the code can be properly formatted and indented using Prettier.

For details on `--embedded-language-formatting`, see [here](https://prettier.io/blog/2020/08/24/2.1.0.html) and [here](https://github.com/thorn0/prettier/blob/master/src/language-js/embed.js). Unfortunately, it looks like currently Prettier doesn't support `js` code tag directly (see https://github.com/prettier/prettier/pull/5993). The workaround is to use `html` code tag and embed the JavaScript code in HTML `<script>` tag.

> **Note**
> Dear future reader, if you wish to embed TypeScript code, e.g. `useSharedValue<number>(42)`, you will need to specify `lang="ts"` attribute for the `<script>` tag and also modify the implementation of `runPlugin` function.


## Test plan

1. Break formatting in some code input in `__tests__/plugin.test.js`
2. Confirm that `yarn lint:js` fails
3. Run `yarn format:js`
4. Confirm that the file has been formatted properly
5. Break formatting again
6. Stage the changes
7. Confirm that `git commit` fails due to an empty commit
